Sources/Amplitude/Plugins/AnalyticsConnectorPlugin.swift

```diff
@@ -8,8 +8,10 @@ class AnalyticsConnectorPlugin: BeforePlugin {
     override func setup(amplitude: Amplitude) {
         super.setup(amplitude: amplitude)
         connector = AnalyticsConnector.getInstance(amplitude.configuration.instanceName)
-        connector!.eventBridge.setEventReceiver { event in
-            amplitude.track(event: BaseEvent(
+        let logger = amplitude.logger
+        connector!.eventBridge.setEventReceiver { [weak amplitude, logger] event in
+            logger?.error(message: "Amplitude instance has been deallocated, please maintain a strong reference to track events from Experiment")
```

Comment:
@izaaz added an error message if we start to receive events from Experiment but the amplitude instance was deallocated. This does maintain a strong reference to logger, but this should be better than the entire amplitude instance.